app/Platform/Models/Game.php

```diff
@@ -161,6 +161,28 @@ public function shouldBeSearchable(): bool
         return false;
     }
 
+    // == temp subset detection
+
+    public function getParentGameId(): ?int
+    {
+        // Use regular expression to check if the title includes a subset pattern and extract the base title.
+        if (preg_match('/(.+)\[Subset - .+\]/', $this->Title, $matches)) {
+            // Trim to ensure no leading/trailing spaces.
+            $baseSetTitle = trim($matches[1]);
+
+            // Attempt to find a game with the base title and the same console ID.
+            $game = Game::where('Title', $baseSetTitle)
+                 ->where('ConsoleID', $this->ConsoleID)
+                 ->first();
```

Comment:
```suggestion
                ->where('ConsoleID', $this->ConsoleID)
                ->first();
```

nit: There's some extra whitespace here that should be removed to keep with 4-space indentation.